docs/fastapi.rst

```diff
@@ -0,0 +1,320 @@
+FastAPI
+=======
+
+The package ``dockerflow.fastapi`` package implements various tools to support
+`FastAPI`_ based projects that want to follow the Dockerflow specs:
+
+- A Python logging formatter following the `mozlog`_ format.
+
+- A Flask extension implements:
+
+  - Emitting of `request.summary`_ log records based on request specific data.
+
+  - Views for health monitoring:
+
+    - ``/__version__`` - Serves a ``version.json`` file
+
+    - ``/__heartbeat__`` - Runs the configured Dockerflow checks
+
+    - ``/__lbheartbeat__`` - Retuns a HTTP 200 response
+
+  - Hooks to add custom Dockerflow checks.
+
+.. _`FastAPI`: https://fastapi.tiangolo.com
+.. _`mozlog`: https://github.com/mozilla-services/Dockerflow/blob/main/docs/mozlog.md
+.. _`request.summary`: https://github.com/mozilla-services/Dockerflow/blob/main/docs/mozlog.md#application-request-summary-type-requestsummary
+
+.. seealso::
+
+    For more information see the :doc:`API documentation <api/fastapi>` for
+    the ``dockerflow.fastapi`` module.
+
+Setup
+-----
+
+To install ``python-dockerflow``'s FastAPI support please follow these steps:
+
+#. In your code where your FastAPI application lives set up the dockerflow FastAPI
+   extension::
+
+     from fastapi import FastAPI
+     from dockerflow.fastapi import router
+     from dockerflow.fastapi.middleware import MozlogRequestSummaryLogger
+
+     app = FastAPI()
+     app.include_router(router)
+     app.add_middleware(MozlogRequestSummaryLogger)
+
+#. Make sure the app root path is set correctly as this will be used
+   to locate the ``version.json`` file that is generated by
+   your CI or another process during deployment.
+
+   .. seealso:: :ref:`fastapi-versions` for more information
+
+#. Configure logging to use the ``JsonLogFormatter`` logging formatter for the
+   ``request.summary`` logger (you may have to extend your existing logging
+   configuration), see :ref:`fastapi-logging` for more information.
+
+.. _fastapi-config:
+
+Configuration
+-------------
+
+.. epigraph::
+
+   Accept its configuration through environment variables.
```

Comment:
Does `epigraph` suggest that this should have more formatting that just an indentation?
<img width="446" alt="image" src="https://github.com/mozilla-services/python-dockerflow/assets/20747904/72e6655f-8ceb-4a9a-ba94-d89b59cf7efa">

Reply:
This is like a note... I'd say that it just has some indentation